### ISSUE TITLE
Create many to many instance with id?

### ISSUE THREAD

**Issue by seanchenca** [typeorm/typeorm#1241]
Hi,

Let say I have two classes:

```
@Entity('resource')
class Resource extends EntBase {

  @Column({ type: 'varchar', length: 64 })
  title: string

  @ManyToMany(type => Topic, topic => topic.resources)
  @JoinTable({ name: 'resource_topic' })
  topics: Topic[]
}

@Entity('topic')
class Topic extends EntBase {

  @Column({ type: 'varchar', length: 64 })
  topic: string

  @ManyToMany(type => Resource, resource => resource.topics)
  resources: Resource[]
}
```

and I would like to create a new Resource with two topics, and I already know the ids of the two topic (suppose they are 1, and 2), I am assuming I have to do:

```
const topic1 = awiat topicRepo.findById(1);
resource.topics.push(topic1);
const topic2 = await topicRepo.findById(2);
resource.topics.push(topic2);
resourceRepo.create(resource);
```

 is it possible to save the Resource without creating instances of Topic? or do I have to manipulate the 'resource_topic' table manually?

btw, if I omit the {type: 'varchar', length: 64} for the topic column and leave it just:
```
@Column()
topic: string
```

I got an error:

```
sqlMessage: 'You have an error in your SQL syntax; check the manual that corresponds to yourMySQL server version for the right syntax to use near \'NOT NULL, `lang` enum(\'en\', \'fr\') NOT NULL) ENGINE=InnoDB\' at line 1',
[Node]   sqlState: '42000',
[Node]   index: 0,
[Node]   sql: 'CREATE TABLE `topic` (`id` int(11) NOT NULL PRIMARY KEY AUTO_INCREMENT, `createdAt` datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6), `updatedAt` datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6), `version` int(11) NOT NULL, `topic`  NOT NULL, `lang` enum(\'en\', \'fr\') NOT NULL) ENGINE=InnoDB',
```

looks like the query is not specifying type of `topic`, I thought typeorm is going to infer that from the string type and make it varchar(255), am I missing some settings?

**Comment by pleerock** [comment]
Regarding to your question - checkout how to work with [relational query builder](http://typeorm.io/#/relational-query-builder).

Regarding to your error - did you specify a reflect-metadata properly? Looks like it cannot get type reflection from your properties.

**Comment by seanchenca** [comment]
thanks @pleerock for pointing out the doc. works perfectly:
```getConnection()
      .createQueryBuilder()
      .relation(Resource, 'topics')
      .of(resourceId)
      .addAndRemove([...], [...])
```

for the error, do you refer to the option in tsconfig.json?:
```
"emitDecoratorMetadata": true,
```
I do have this option enabled, or is it something else?

**Comment by seanchenca** [comment]
hi @pleerock, seems I have some problem with the addAndRemove call. let say originally I have two resources id 1 and 2. resource 1 has topics id 1 and 2, resource 2 has topics 3 and 4. so the resource_topic table has the following record:
```
resourceId  |  topicId
----------------------
       1    |      1
       1    |      2
       2    |      3
       2    |      4
```
now I want to update resource 2 to have topics id 5 and 6, after I call:
```
getConnection()
      .createQueryBuilder()
      .relation(Resource, 'topics')
      .of(2)
      .addAndRemove([5, 6], [3, 4]);
```
the resource_topic table become:
```
resourceId  |  topicId
----------------------
       2    |      5
       2    |      6
```
i.e. the record for resource id 1 are wiped out too. could you see what I did wrong here?

**Comment by pleerock** [comment]
Its not expected behaviour. If you send empty array it should not do anything. Fixed and will be released in 0.1.7
